src/backend/crud/agent.py

```diff
@@ -277,23 +278,23 @@ def update_agent(
 @validate_transaction
 def delete_agent(db: Session, agent_id: str, user_id: str) -> bool:
     """
-    Delete an agent by ID.
-    Anyone can delete a public agent, but only the owner can delete a private agent.
+    Delete an Agent by ID if the Agent was created by the user_id given.
 
     Args:
         db (Session): Database session.
         agent_id (str): Agent ID.
 
     Returns:
-      bool: True if the agent was deleted, False otherwise
+      bool: True if the Agent was deleted, False otherwise
     """
     agent_query = db.query(Agent).filter(Agent.id == agent_id)
     agent = agent_query.first()
 
     if not agent:
         return False
 
-    if agent and agent.is_private and agent.user_id != user_id:
+    # Can only delete Agent created by user_id
+    if agent.user_id != user_id:
```

Comment:
Instead of checking if the user_id is the same here, we could add the `user_id` to the query. 

`agent_query = db.query(Agent).filter(Agent.id == agent_id, Agent.user_id == user_id)`

Reply:
Good point, will do